core/src/test/scala/caliban/wrappers/CostEstimationSpec.scala

```diff
@@ -0,0 +1,155 @@
+package caliban.wrappers
+
+import caliban.CalibanError.ValidationError
+import caliban.GraphQL.graphQL
+import caliban.Macros.gqldoc
+import caliban.ResponseValue.ObjectValue
+import caliban.RootResolver
+import caliban.Value.{ FloatValue, IntValue }
+import caliban.schema.Annotations.GQLDirective
+import caliban.wrappers.CostEstimation.CostDirective
+import zio.console.putStr
+import zio.test.environment.TestConsole
+import zio.test.{ assertTrue, DefaultRunnableSpec, ZSpec }
+import zio.{ Ref, UIO }
+
+object CostEstimationSpec extends DefaultRunnableSpec {
+  case class WithArgs(limit: Int)
+  case class E(value: Int)
+  @GQLDirective(CostDirective(10))
+  case class D(e: E, @GQLDirective(CostDirective(1, List("limit"))) f: List[String])
+  case class Test(
+    a: Int,
+    @GQLDirective(CostDirective(100)) b: UIO[Int],
+    @GQLDirective(CostDirective(5, multipliers = List("limit"))) c: WithArgs => UIO[List[Int]],
```

Comment:
Didn't completely get the multipliers until I saw this, this is really nice.

Reply:
Yeah I had another idea that I saw ibm uses which is to have the cost field be a string that can be coerced into a formula for computing the cost of the field, i.e. `5 * limit`, but parsing that seemed like it might be a bit difficult (if we allowed floats for instance) so I went for the easy path instead.